mipengine/controller/node_landscape_aggregator.py

```diff
@@ -23,107 +22,80 @@
 logger = ctrl_logger.get_background_service_logger()
 
 NODE_LANDSCAPE_AGGREGATOR_REQUEST_ID = "NODE_LANDSCAPE_AGGREGATOR"
-GET_NODE_INFO_SIGNATURE = "mipengine.node.tasks.common.get_node_info"
-GET_NODE_DATASETS_PER_DATA_MODEL_SIGNATURE = (
-    "mipengine.node.tasks.common.get_node_datasets_per_data_model"
-)
-GET_DATA_MODEL_CDES_SIGNATURE = "mipengine.node.tasks.common.get_data_model_cdes"
 NODE_LANDSCAPE_AGGREGATOR_UPDATE_INTERVAL = (
     controller_config.node_landscape_aggregator_update_interval
 )
 CELERY_TASKS_TIMEOUT = controller_config.rabbitmq.celery_tasks_timeout
 
 
-async def _get_nodes_info(nodes_socket_addr: List[str]) -> List[NodeInfo]:
-    celery_apps = [
-        get_node_celery_app(socket_addr) for socket_addr in nodes_socket_addr
-    ]
-    nodes_task_signature = {
-        celery_app: celery_app.signature(GET_NODE_INFO_SIGNATURE)
-        for celery_app in celery_apps
-    }
-
-    tasks_coroutines = [
-        _task_to_async(task, app=app)(request_id=NODE_LANDSCAPE_AGGREGATOR_REQUEST_ID)
-        for app, task in nodes_task_signature.items()
-    ]
-    results = await asyncio.gather(*tasks_coroutines, return_exceptions=True)
-    nodes_info = [
-        NodeInfo.parse_raw(result)
-        for result in results
-        if not isinstance(result, Exception)
+def _get_nodes_info(nodes_socket_addr: List[str]) -> List[NodeInfo]:
+    node_info_tasks_handlers = [
+        NodeInfoTasksHandler(
+            node_queue_addr=node_socket_addr, tasks_timeout=CELERY_TASKS_TIMEOUT
+        )
+        for node_socket_addr in nodes_socket_addr
     ]
 
-    for app in celery_apps:
-        app.close()
+    async_results = {}
+    for tasks_handler in node_info_tasks_handlers:
+        try:
+            tmp = tasks_handler.queue_node_info_task(
+                request_id=NODE_LANDSCAPE_AGGREGATOR_REQUEST_ID
+            )
+            async_results[tasks_handler] = tmp
+        except (CeleryConnectionError, CeleryTaskTimeoutException) as exc:
```

Comment:
I m not sure what you mean "All Exceptions should be cought here", buy if an exception other than `CeleryConnectionError` or `CeleryTaskTimeoutException` is raised here, it will be handled by the "catch-all" block in `_update`

Reply:
The main loop (the while loop in `_update`) would not stop. What you probably mean is that if an exception other than `CeleryConnectionError` or `CeleryTaskTimeoutException` is raised inside `_get_nodes_info`, `_get_node_datasets_per_data_model` or `_get_node_cdes` then the update of **all** the nodes will be "skipped" for this iteration of the loop.
Anyhow, I added a second generic catch block, on each of the aforementioned functions, that will log any other exception types raised there and prohibit them from bubbling higher up.